### PR TITLE
WRN-5319: Add unit tests to FormCheckboxItem.

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
There were no tests for FormCheckboxItem.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Created tests for FormCheckboxItem.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-5319

### CODE REVIEW

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import '@testing-library/jest-dom';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import FormCheckboxItem, {FormCheckboxItemBase} from '../FormCheckboxItem';
+import Icon from '../../Icon';
+
+describe('CheckboxItem Specs', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:8]: Please change to FormCheckboxItem specs.

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	});
+
+	test('should have correct text', () => {
+		render(<FormCheckboxItemBase>Hello FormCheckboxItem</FormCheckboxItemBase>);
+		const element = screen.getByText(/Hello FormCheckboxItem/i);
+
+		expect(element).toBeInTheDocument();
+	});
+
+	test('should select with click', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:26]:
```suggestion
	test('should select by clicking twice', () => {
```
Just a suggestion.

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		const formCheckboxItemElement = screen.getAllByRole('checkbox')[0];
+
+		const expected = 'selected';
+
+		userEvent.click(formCheckboxItemElement);
+
+		expect(formCheckboxItemElement).toHaveClass(expected);
+	});
+
+	test('should deselect with click', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:37]:
```suggestion
	test('should deselect by clicking twice', () => {
```
Just a suggestion.

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+		const expected = 'selected';
+
+		userEvent.click(formCheckboxItemElement);
+		userEvent.click(formCheckboxItemElement);
+
+		expect(formCheckboxItemElement).not.toHaveClass(expected);
+	});
+
+	test('should be disabled when it has prop `disabled`', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:49]:
```suggestion
	test('should be disabled when `disabled` prop is true', () => {
```

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+		render(<FormCheckboxItem disabled>Hello FormCheckboxItem</FormCheckboxItem>);
+		const formCheckboxItemElement = screen.getAllByRole('checkbox')[0];
+
+		const expected = 'disabled';
+
+		expect(formCheckboxItemElement).toHaveAttribute(expected);
+	});
+
+	test('should be have the icon `-` when indeterminate', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:58]:
```suggestion
	test('should have the `-` icon when indeterminate', () => {
```

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+		render(<FormCheckboxItem indeterminate>Hello FormCheckboxItem</FormCheckboxItem>);
+		const formCheckboxItemIcon = screen.getAllByRole('checkbox')[1];
+
+		const expected = '-';
+
+		expect(formCheckboxItemIcon).toHaveTextContent(expected);
+	});
+
+	test('should be have the icon `-` when indeterminate even when clicked', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:67]:
```suggestion
	test('should have the `-` icon when indeterminate even when clicked', () => {
```

File: FormCheckboxItem/tests/FormCheckboxItem-specs.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	test('should support slotBefore', () => {
+		render(<FormCheckboxItem slotBefore={<Icon>minus</Icon>}>Hello FormCheckboxItem</FormCheckboxItem>);
+		const formCheckboxItemElement = screen.getAllByRole('checkbox')[0].children.item(1).children;
+
+		const expected = 2;
+
+		expect(formCheckboxItemElement).toHaveLength(expected);
+	});
+
+	test('should have inline class when it has prop `inline`', () => {
```
Comment (stanca-pop-lgp) [inline FormCheckboxItem/tests/FormCheckboxItem-specs.js:104]:
```suggestion
	test('should have inline class when `inline` prop is true', () => {
```

**Review by stanca-pop-lgp — CHANGES_REQUESTED** [review]
Please check out my comments.

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Review by daniel-stoian-lgp — APPROVED** [review]

**Pull request merged**